subgraphs/_reference_/src/common/utils/logger.ts

```diff
@@ -1,26 +1,50 @@
 import { ethereum, log } from "@graphprotocol/graph-ts";
 
+/**
+ * Logger class is used to log messages to be picked up by our logging system.
+ *
+ * @param event The event that triggered the log message
+ * @param funcName The name of the function that triggered the log message
+ * @param msg The message to log (add `{}` to the message to add additional arguments)
+ * @param args Any additional arguments to add to the msg (same format as graph-ts log functions)
+ */
 export class Logger {
   event: ethereum.Event;
+  fnName: string | null;
 
-  constructor(event: ethereum.Event) {
+  constructor(event: ethereum.Event, fnName: string | null = null) {
     this.event = event;
+    this.fnName = fnName;
   }
 
-  private formatLog(funcName: string, msg: string): string {
-    return `messari_log: [${funcName}}] tx: ${this.event.transaction.hash.toHexString()} :: ${msg}`;
+  private formatLog(msg: string): string {
+    if (this.fnName) {
+      return `messari_log: [${
+        this.fnName
+      }}] tx: ${this.event.transaction.hash.toHexString()} :: ${msg}`;
+    }
+
+    return `messari_log: tx: ${this.event.transaction.hash.toHexString()} :: ${msg}`;
   }
 
-  info(funcName: string, msg: string): void {
-    log.info(this.formatLog(funcName, msg), []);
+  info(msg: string, args: string[] | null = null): void {
+    if (!args) args = [];
```

Comment:
Instead of using `null` as the default, do `[]`. This way you don't need to check `if(!args)` every time? :P